core/thread/src/TThreadExecutor.cxx

```diff
@@ -1,15 +1,32 @@
 #include "ROOT/TThreadExecutor.hxx"
 #include "tbb/tbb.h"
+#include "TROOT.h"
+#include <iostream>
 
 namespace ROOT{
+
+  unsigned TThreadExecutor::fPoolSize = 0;
+
   TThreadExecutor::TThreadExecutor():fInitTBB(new tbb::task_scheduler_init()){
   }
 
   TThreadExecutor::TThreadExecutor(size_t nThreads):fInitTBB(new tbb::task_scheduler_init(nThreads)){
+    if (fPoolSize != 0){
```

Comment:
What about
```
TThreadExecutor exe1;
TThreadExecutor exe2(12);
```
Shouldn't this also warn? `fPoolSize` will still be at 0 when `exe2` is constructed... Can you get the "current pool size" from TBB instead of keeping a local copy?